tests/TestCase/Routing/Route/RouteTest.php

```diff
@@ -142,6 +142,85 @@ public function testRouteBuildingSmallPlaceholders()
         $this->assertEquals('/fighters/123/move/8/42', $result);
     }
 
+    /**
+     * Test route compile with brace format.
+     *
+     * @return void
+     */
+    public function testRouteCompileBraces()
+    {
+        $route = new Route(
+            '/fighters/{id}/move/{x}/{y}',
+            ['controller' => 'Fighters', 'action' => 'move'],
+            ['id' => '\d+', 'x' => '\d+', 'y' => '\d+', 'pass' => ['id', 'x', 'y']]
+        );
+        $pattern = $route->compile();
+        $this->assertRegExp($pattern, '/fighters/123/move/8/42');
+
+        $result = $route->match([
+            'controller' => 'Fighters',
+            'action' => 'move',
+            'id' => 123,
+            'x' => 8,
+            'y' => 42
+        ]);
+        $this->assertEquals('/fighters/123/move/8/42', $result);
+
+        $route = new Route(
+            '/images/{id}/{x}x{y}',
```

Comment:
hi
can you add a test for :point_down: ?
```php
{ id}
{id }
```

Reply:
Why? `{ id}` shouldn't work. :)

Reply:
I can add a negative test showing that it *doesn't* work. It might also be wise to have tests for `A-Z` and `_`.

Reply:
Done!

Reply:
why `i_d` and `id99` is valid ?
this can make confusing with 
```php
//id=123
'/fighters/{id99}' // returns '/fighters/12399' 
```
now 
`99` is valid character but space no